### PR TITLE
Revamp CONTRIBUTING.md guide

### PR DESCRIPTION
**Pull request by pedroerp**
Refactoring and adding more context to the CONTRIBUTING guide to reflect the current guidelines.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,111 +1,190 @@
...
+   please start a separate discussion with the community to update the
+   coding style document first.
+   * If you are simply updating code that did not comply with the coding 
+   style, please do so in a standalone PR isolated from other logic changes. 
+
+2. **Small Incremental Changes**: If the change large, work with the maintainers
+   on a plan to break and submit it as smaller (yet atomic) parts.
+   * [Research indicates](https://smartbear.com/learn/code-review/best-practices-for-peer-code-review/) 
+   that engineers can only effectively review up to 
+   400 LOC at a time. The human brain can only process so much information
```
Comment (mbasmanova) [inline CONTRIBUTING.md:115]: nit: LOC -> lines of code

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,111 +1,190 @@
...
+
+4. **Code Comments**: Appropriately add comments to your code and document APIs.
+   * As a library, Velox code should be optimized for the reader, not the
+   writer.
+   * Comments should capture information that was on the writer’s mind, but
+   couldn’t be represented as code. 
+   * The overall goal is to make the code more obvious and remove obscurity.
+   * As a guideline, every file, class, member variable, and member function
+   that is not a getter/setter should be documented.
+   * For functions with large bodies, a good practice is to group blocks of
```
Comment (mbasmanova) [inline CONTRIBUTING.md:142]: nit: perhaps, mention that such functions should be rare and suggest to break up the function first.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,111 +1,190 @@
...
+   be deliberately created. 
+   * As a rule of thumb, components should be deep and encapsulate as much
+   complexity as possible, and APIs should be narrow, minimizing dependencies
+   across components and preventing implementation details from leaking through
+   the API.
+
+## Adding Scalar Functions
+
+Adding Presto and Spark functions are a good way to get yourself familiar with
+Velox and the coding review process. In addition to the general contribution
```
Comment (mbasmanova) [inline CONTRIBUTING.md:161]: coding -> code

**Review by mbasmanova — APPROVED** [review]
Thanks.

**Review by majetideepak — COMMENTED** [review]
@pedroerp changes look nice. Some minor comments.
I notice that contributors `Resolve conversation` the reviewer's comments after addressing them.
I feel it should be the reviewer who should Resolve their comments after verifying.
It will be great if you can clarify this.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,111 +1,190 @@
...
+1. **Coding Style**: Review and strictly follow our coding style document, 
+   available in [`CODING_STYLE.md`](CODING_STYLE.md).
+   * Velox favors consistency over personal preference. If there are 
+   technical reasons why a specific guideline should not be followed, 
+   please start a separate discussion with the community to update the
+   coding style document first.
+   * If you are simply updating code that did not comply with the coding 
+   style, please do so in a standalone PR isolated from other logic changes. 
+
+2. **Small Incremental Changes**: If the change large, work with the maintainers
```
Comment (aditi-pandit) [inline CONTRIBUTING.md:111]: Nit : missed is "If the change is large"

**Review by aditi-pandit — APPROVED** [review]
Thanks Pedro. LGTM.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,111 +1,190 @@
...
+5. Make sure that obvious inefficiencies are addressed. 
+   * If appropriate, provide micro-benchmarks to support your claims with data. 
 
-6. Add tests.
+4. Add documentation for the new function to an .rst file under velox/docs/functions directory.
+   * Functions in documentation are listed in alphabetical order. Make sure to
+   place the new function so that the order is preserved.
 
-7. Run [Fuzzer](https://facebookincubator.github.io/velox/develop/testing/fuzzer.html)
+6. Run [Fuzzer](https://facebookincubator.github.io/velox/develop/testing/fuzzer.html)
```
Comment (kgpai) [inline CONTRIBUTING.md:187]: nit: Should we add a section about creating or checking up for fuzzer issues as the fuzzer uncovers problems (also adding a note upload fuzzer artifacts).

Comment (pedroerp) [reply CONTRIBUTING.md:187]: Is this different from the information we have here?

https://facebookincubator.github.io/velox/develop/testing/fuzzer.html

let's sync up offline and discuss what information would make sense to add here.

**Comment by pedroerp** [comment]
Thanks everyone for the feedback/review. All comments have been addressed. 

> I notice that contributors `Resolve conversation` the reviewer's comments after addressing them.
> I feel it should be the reviewer who should Resolve the conversation after verifying.
> It will be great if you can clarify this.

Good point. I added an item to clarify this.

**Pull request closed without merging**